multicluster/controllers/multicluster/member_clusterset_controller.go

```diff
@@ -290,3 +311,36 @@ func (r *MemberClusterSetReconciler) updateStatus() {
 		klog.ErrorS(err, "Failed to update Status of ClusterSet", "Name", namespacedName)
 	}
 }
+
+// SetRemoteCommonAreaManager is for testing only
+func (r *MemberClusterSetReconciler) SetRemoteCommonAreaManager(mgr commonarea.RemoteCommonAreaManager) commonarea.RemoteCommonAreaManager {
+	r.remoteCommonAreaManager = mgr
+	return r.remoteCommonAreaManager
+}
+
+func (r *MemberClusterSetReconciler) GetRemoteCommonAreaAndLocalID() (commonarea.RemoteCommonArea, string, error) {
+	var rwm sync.RWMutex
+	rwm.RLock()
+	defer rwm.RUnlock()
```

Comment:
Which race condition can it prevent if it declares a new mutex every time the read method is called?
I think the race condition between `Reconcile` and `GetRemoteCommonAreaAndLocalID` still exist

Reply:
Fixed, I misunderstood the usage of RWMutex, refine the mutex in MemberClusterSetReconciler to RWMutex. thanks.